Update HTTP -> HTTPS

him2him2 opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
URLs updated to use HTTPS protocol where appropriate to improve security and privacy.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- Docs Pull Request


##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
Update to docs.

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

Commit:
Update HTTP -> HTTPS

URLs updated to use HTTPS protocol where appropriate to improve security and privacy.

aminvakil commented on docs/docsite/rst/user_guide/playbooks_variables.rst:
```diff
@@ -188,7 +188,7 @@ To reference an IP address from your facts using the dot notation::
 ==========================================
 
-Jinja2 filters let you transform the value of a variable within a template expression. For example, the ``capitalize`` filter capitalizes any value passed to it; the ``to_yaml`` and ``to_json`` filters change the format of your variable values. Jinja2 includes many `built-in filters <http://jinja.pocoo.org/docs/templates/#builtin-filters>`_ and Ansible supplies many more filters. To find more exa ...
+Jinja2 filters let you transform the value of a variable within a template expression. For example, the ``capitalize`` filter capitalizes any value passed to it; the ``to_yaml`` and ``to_json`` filters change the format of your variable values. Jinja2 includes many `built-in filters <https://jinja.palletsprojects.com/en/2.11.x/templates/#builtin-filters>`_ and Ansible supplies many more filters.  ...
```
Wouldn't it be clearer if the version wouldn't be specified so that there isn't a need to bump the version once in a while?
```suggestion
Jinja2 filters let you transform the value of a variable within a template expression. For example, the ``capitalize`` filter capitalizes any value passed to it; the ``to_yaml`` and ``to_json`` filters change the format of your variable values. Jinja2 includes many `built-in filters <https://jinja.palletsprojects.com/templates/#builtin-filters>`_ and Ansible supplies many more filters. To find more examples of filters, see :ref:`playbooks_filters`.
```

him2him2 replied:
It would, Looks like https://jinja.palletsprojects.com/templates/#builtin-filters also provides a 304 to the current version and language. 

I have changed it and committed the change.

Commit:
Modification to HTTPS updates for https://jinja.palletsprojects.com

I removed the version and language from the link. The links will produce a 304 to the language and current version.

samccann commented on docs/docsite/rst/community/documentation_contributions.rst:
```diff
@@ -6,7 +6,7 @@ Contributing to the Ansible Documentation
 Ansible has a lot of documentation and a small team of writers. Community support helps us keep up with new features, fixes, and changes.
 
-Improving the documentation is an easy way to make your first contribution to the Ansible project. You do not have to be a programmer, since most of our documentation is written in YAML (module documentation) or `reStructuredText <http://docutils.sourceforge.net/rst.html>`_ (rST). Some collection-level documentation is written in a subset of `Markdown <https://github.com/ansible/ansible/issues/68 ...
+Improving the documentation is an easy way to make your first contribution to the Ansible project. You do not have to be a programmer, since most of our documentation is written in YAML (module documentation) or `reStructuredText <https://docutils.sourceforge.net/rst.html>`_ (rST). Some collection-level documentation is written in a subset of `Markdown <https://github.com/ansible/ansible/issues/6 ...
```
```suggestion
Improving the documentation is an easy way to make your first contribution to the Ansible project. You do not have to be a programmer, since most of our documentation is written in YAML (module documentation) or `reStructuredText <https://docutils.sourceforge.io/rst.html>`_ (rST). Some collection-level documentation is written in a subset of `Markdown <https://github.com/ansible/ansible/issues/68119#issuecomment-596723053>`_. If you are using Ansible, you already use YAML in your playbooks. rST and Markdown are mostly just text. You do not even need git experience, if you use the ``Edit on GitHub`` option.
```

samccann commented on docs/docsite/rst/dev_guide/style_guide/index.rst:
```diff
@@ -238,7 +238,7 @@ Modules require different suffixes from other plugins:
 ^^^^^^^^^^^^^^^^^
 
-The page you're reading includes a `local TOC <http://docutils.sourceforge.net/docs/ref/rst/directives.html#table-of-contents>`_.
+The page you're reading includes a `local TOC <https://docutils.sourceforge.net/docs/ref/rst/directives.html#table-of-contents>`_.
```
```suggestion
The page you're reading includes a `local TOC <https://docutils.sourceforge.io/docs/ref/rst/directives.html#table-of-contents>`_.
```

him2him2 replied:
I see the 304 redirect. As requested, updated those urls and committed the changes. Thank you!

samccann reviewed (commented):
the docutils .net urls don't work for me but the .io versions do.

Thanks for this PR!

Commit:
Updated links for  https://docutils.sourceforge.net

As requested, updated links for http://docutils.sourceforge.net ->  https://docutils.sourceforge.io

samccann commented:
Thanks @him2him2 for the Ansible docs PR fix!
